sklearn/utils/estimator_checks.py

```diff
@@ -133,6 +133,40 @@ def _yield_classifier_checks(name, classifier):
     yield check_decision_proba_consistency
 
 
+def _yield_classifier_classes_param_checks(name, estimator):
+    # to be moved to _yield_classifier_checks once classes added to all
+    # classifiers
+    estimators_to_check = set(['GaussianNB',
+                               'BernoulliNB',
+                               'MultinomialNB',
+                               'DecisionTreeClassifier',
+                               'ExtraTreeClassifier',
+                               'RandomForestClassifier',
+                               'ExtraTreesClassifier',
+                               'GradientBoostingClassifier',
+                               'LinearSVC',
+                               'LogisticRegression',
+                               'LogisticRegressionCV'
+                               ])
+    estimators_multilabel = set(['DecisionTreeClassifier',
+                                 'ExtraTreeClassifier',
+                                 'RandomForestClassifier',
+                                 'ExtraTreesClassifier'])
+
+    if name in estimators_to_check:
+        yield check_classes_parameter_sorted_unique
+
+        if hasattr(estimator, 'predict_proba'):
+            yield check_classes_parameter_predict_proba
```

Comment:
I'm not sure why you want a separate test for `predict_proba` and `decision_function`. Can they be merged? You also need to check that the value of `classes` is copied to `classes_` when present.

Reply:
there are a couple of reasons why:
1. in a binary classification case, predict_proba returns 2 vectors and decision function only 1.
2. when passing extra classes, not present in the data, then probabilities for those classes should be 0, but what should be the decision function for those classes? it depends on the estimator I suppose so the checks are slightly different for both cases.

i can create if else structure in the check, but i thought it might be cleaner code to keep them separate.

i'll add the check you suggested. we do something similar in scorer but it should be here as well.